code/modules/food_and_drinks/recipes/tablecraft/recipes_frozen.dm

```diff
@@ -229,3 +241,48 @@
 	)
 	result = /obj/item/reagent_containers/food/snacks/snowcones/rainbow
 	subcategory = CAT_ICE
+
+//////////////////////////POPSICLES///////////////////////
+
+// This section includes any frozen treat served on a stick.
+////////////////////////////////////////////////////////////
+
+/datum/crafting_recipe/food/orange_popsicle
+	name = "Orange popsicle"
+	reqs = list(
+		/obj/item/popsicle_stick = 1,
+		/datum/reagent/consumable/orangejuice = 4,
+		/datum/reagent/consumable/ice = 2,
+		/datum/reagent/consumable/cream = 2,
+		/datum/reagent/consumable/vanilla = 2,
+		/datum/reagent/consumable/sugar = 2,
+	)
+	result = /obj/item/reagent_containers/food/snacks/popsicle/creamsicle_orange
+	subcategory = CAT_ICE
+
+/datum/crafting_recipe/food/berry_popsicle
+	name = "Berry popsicle"
+	reqs = list(
+		/obj/item/popsicle_stick = 1,
+		/datum/reagent/consumable/berryjuice = 4,
+		/datum/reagent/consumable/ice = 2,
+		/datum/reagent/consumable/cream = 2,
+		/datum/reagent/consumable/vanilla = 2,
+		/datum/reagent/consumable/sugar = 2,
+	)
+	result = /obj/item/reagent_containers/food/snacks/popsicle/creamsicle_berry
+	subcategory = CAT_ICE
+
+/datum/crafting_recipe/food/orange_popsicle
```

Comment:
you have orange popsicle repeat 2 times

Reply:
you also forgot to make a recipe for the popsicle i cannot name or i will get banished to the shadow realm

Reply:
Good catch. I will do some more testing today to make sure everything works.

I must admit i didn't include the nogga black recipe because i wanted to save it as a rare drop from once of the freezer types in my upcoming loot crate PR.

Since we don't have a ammonium chloride reagent, should i substitute it with ammonia + chlorine, or blumpkin juice? 

Blumpkin juice has implied ammonium chloride so it might be a good choice. Blumpkin juice, black crayon powder, ice, cream, sugar, something like that.